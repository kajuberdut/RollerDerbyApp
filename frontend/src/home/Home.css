
.Home {
    width: 43%;
    margin-top: 200px;
    margin-left: 12%; 
}

.Home-Title {
  padding: 20px;
}

.Home-Link {
    padding: 5px;
}
