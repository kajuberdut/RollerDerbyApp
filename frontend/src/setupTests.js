// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import './setupTests.js';

// https://stackoverflow.com/questions/32911630/how-do-i-deal-with-localstorage-in-jest-tests
const localStorageMock = {
    getItem: jest.fn(),
    setItem: jest.fn(),
    clear: jest.fn()
  };
  global.localStorage = localStorageMock;