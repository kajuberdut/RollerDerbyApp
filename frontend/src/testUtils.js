import React from "react";
import UserContext from "./multiUse/UserContext";

const demoUser = {
    userId: 1,
    username: "testUser",
    email: "testuser@gmail.com",
    facebookName: null,
    about: null,
    primaryNumber: null,
    level: null,
    ruleset: [],
    position: [],
    insurance: [],
    locationId: null,
    associatedLeagues: null,
    phoneNumber: null,
    firstName: null,
    lastName: null,
    additionalInfo: null,
    secondaryNumber: null
  }

const UserProvider =
    ({ children, user = demoUser }) => (
    <UserContext.Provider value={{ user }}>
      {children}
    </UserContext.Provider>
);



export { UserProvider };
