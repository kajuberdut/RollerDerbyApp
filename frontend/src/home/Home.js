import React, { useEffect } from "react";
import './Home.css'
import { Link } from "react-router-dom";
import { Card, CardBody, CardText, CardTitle, Button } from "reactstrap";

/**
 * Display home page
 */

function Home({setIsHomeVis}) {

  /** Get user from context*/

  const user = JSON.parse(localStorage.getItem('user'));

  /** When home page is mounted setIsloginVis to true - false when unmounted */

  useEffect(() => {
    setIsHomeVis(true);
  return () => {
    setIsHomeVis(false);
  };
  }, []);

  /** Render home page */

    return (
      <section className="col-md-8 Home">
        <Card>
          <CardTitle className="Home-Title">
              <h1>The block party begins when the star arrives.</h1>
          </CardTitle>
          <CardBody >
            {user && <h2>Welcome to the party, {user.username}!</h2>}
            {!user && <CardText>
              <Link className="Home-Link" to="/login"><Button>Login</Button></Link>
              <Link className="Home-Link" to="/signup"><Button>Signup</Button></Link>
            </CardText>
            }
          </CardBody>
        </Card>
      </section>
    )
}

export default Home;
