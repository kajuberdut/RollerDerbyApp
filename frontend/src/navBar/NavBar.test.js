import { render, screen, cleanup, fireEvent  } from '@testing-library/react';
import React from 'react';
import Navbar from './NavBar';
import { Nav } from 'reactstrap';
import { UserProvider } from "../testUtils";
import { BrowserRouter } from "react-router-dom";
import AllRoutes from '../routes/Routes.js';
import Home from '../home/Home.js';
// import localStorageMock from 'jest-localstorage-mock';
import '../setupTests.js';

// npm test Navbar.test.js
// must be in frontend directory 

afterEach(() => {
  cleanup()
  // console.log("afterEach is running")
})

describe("user logged in", () => {  

    test('renders NavBar component', () => {
      render(
      <BrowserRouter>
          <UserProvider>
              <Navbar />
          </UserProvider>
      </BrowserRouter>
      );
    });

    test('matches snapshot', function() {
      const { asFragment } = render(  
        <BrowserRouter>
            <UserProvider>
                <Navbar />
            </UserProvider>
        </BrowserRouter>);
      expect(asFragment()).toMatchSnapshot(); 
    })


    test('renders brand and links', () => {
      render(
        <BrowserRouter>
            <UserProvider>
                <Navbar logout={jest.fn()} />
            </UserProvider>
        </BrowserRouter>
      );

      const brandLink = screen.getByText('Conteact');

      expect(brandLink).toBeInTheDocument();
      expect(brandLink).toHaveAttribute('href', '/');

      const profileLink = screen.getByText('testUser');
      expect(profileLink).toBeInTheDocument();
      expect(profileLink).toHaveAttribute('href', '/profile');

    });

    test("it navigates to the bouts page when clicked", () => {
      render(
        // <BrowserRouter initialEntries={["/"]}> 
        <BrowserRouter> 
          <UserProvider >
              <Navbar />
              <AllRoutes>
                <Home />
              </AllRoutes>
          </UserProvider>
        </BrowserRouter>
      );

      expect(screen.queryByText('A Roller Derby Communication Application')).toBeInTheDocument();
      const bouts = screen.getByText("Bouts");
      fireEvent.click(bouts);
      expect(screen.queryByText('A Roller Derby Communication Application')).not.toBeInTheDocument();
      expect(screen.getByText("Create Bout")).toBeInTheDocument();
      expect(window.location.pathname).toBe("/bouts");
   
    });

    test("it navigates to the mixers page when clicked", () => {
      render(

        <BrowserRouter> 
          <UserProvider >
              <Navbar />
              <AllRoutes>
                <Home />
              </AllRoutes>
          </UserProvider>
        </BrowserRouter>
      );

      expect(screen.queryByText('Create Bout')).toBeInTheDocument();
      const mixers = screen.getByText("Mixers");
      fireEvent.click(mixers);
      expect(screen.queryByText('Create Bout')).not.toBeInTheDocument();

      expect(screen.getByText("Create Mixer")).toBeInTheDocument();
      expect(window.location.pathname).toBe("/mixers");
    });

    // ! this one running into issues with mock Api calls
    // test("it navigates to the users page when clicked", () => {
    //   render(

    //     <BrowserRouter> 
    //       <UserProvider >
    //           <Navbar />
    //           <AllRoutes>
    //             <Home />
    //           </AllRoutes>
    //       </UserProvider>
    //     </BrowserRouter>
    //   );
    //   console.log(" ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^window.location.pathname", window.location.pathname)

    //   expect(screen.queryByText('Create Mixer')).toBeInTheDocument();
    //   const user = screen.getByText("testUser");
    //   fireEvent.click(user);
    //   expect(screen.queryByText('Create Mixer')).not.toBeInTheDocument();

    //   expect(screen.getByText("testUser")).toBeInTheDocument();
    //   expect(window.location.pathname).toBe("/profile");
    //   const conteact = screen.getByText("Conteact");
    //   fireEvent.click(conteact);
    // });


});

describe("user logged out", () => {  

  test('renders login link when no user is present', () => {
    render(
      <BrowserRouter>
        <UserProvider user={null}> {/* Provide a null value to simulate no user */}
          <Navbar />
        </UserProvider>
      </BrowserRouter>
    );

    const loginLink = screen.getByText('Login'); 

    expect(loginLink).toBeInTheDocument();
    expect(loginLink).toHaveAttribute('href', '/login');
});

  test('renders signup link when no user is present', () => {
    render(
      <BrowserRouter>
        <UserProvider user={null}> {/* Provide a null value to simulate no user */}
          <Navbar />
        </UserProvider>
      </BrowserRouter>
    );

    const signupLink = screen.getByText('Signup'); 

    expect(signupLink).toBeInTheDocument();
    expect(signupLink).toHaveAttribute('href', '/signup');
  });

});


// beforeEach(() => {
//   localStorage.clear();
// });

// ! right now this is not getting used 

// const userData = {
    
//   user: {
//     userId: 3,
//     username: "SockHer Blue",
//     email: "sockher@gmail.com",
//     facebookName: null,
//     about: null,
//     primaryNumber: null,
//     level: null,
//     ruleset: [],
//     position: [],
//     insurance: [],
//     locationId: null,
//     associatedLeagues: null,
//     phoneNumber: null,
//     firstName: null,
//     lastName: null,
//     additionalInfo: null,
//     secondaryNumber: null
//   }

// }

// localStorage.setItem('user', JSON.stringify(userData));

// // let lsUser = localStorage.getItem('user')
// const lsUser = JSON.parse(localStorage.getItem('user'));
// console.log("!!! lsUser:", lsUser)