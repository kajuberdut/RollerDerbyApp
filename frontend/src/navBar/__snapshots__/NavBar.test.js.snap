// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user logged in matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="NavBar-Top navbar navbar-expand-md"
    >
      <div
        class="container-fluid"
      >
        <div
          style="display: flex;"
        >
          <div
            class="star"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-star "
              data-icon="star"
              data-prefix="fas"
              focusable="false"
              role="img"
              style="font-size: 40px; margin-right: 7px;"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
                fill="currentColor"
              />
            </svg>
          </div>
          <a
            class="navbar-brand"
            href="/"
          >
            Conteact
          </a>
        </div>
        <ul
          class="ml-auto navbar-nav"
        >
          <li
            class="Navbar-Profile nav-item"
          >
            <a
              class=""
              href="/profile"
            >
              testUser
            </a>
          </li>
          <li
            class="Navbar-Users nav-item"
          >
            <a
              class=""
              href="/users"
            >
              Users
            </a>
          </li>
          <li
            class="Navbar-Bouts nav-item"
          >
            <a
              class=""
              href="/bouts"
            >
              Bouts
            </a>
          </li>
          <li
            class="Navbar-Mixers nav-item"
          >
            <a
              class=""
              href="/mixers"
            >
              Mixers
            </a>
          </li>
          <li
            class="Navbar-Logout nav-item"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Logout
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
